app/preprints/-components/preprint-provider-selection/component.ts

```diff
@@ -0,0 +1,39 @@
+import { action } from '@ember/object';
+import { inject as service } from '@ember/service';
+import { serviceLinks } from 'ember-osf-web/const/service-links';
+import { tracked } from '@glimmer/tracking';
+import Component from '@glimmer/component';
+import PreprintProviderModel from 'ember-osf-web/models/preprint-provider';
+import RouterService from '@ember/routing/router-service';
+
+
+interface InputArgs {
+    allProviders: PreprintProviderModel[];
+}
+
+
+export default class PreprintProviderSelection extends Component<InputArgs> {
+    @service router!: RouterService;
+
+    allProviders: PreprintProviderModel[] = this.args.allProviders;
+    learnMoreUrl: string = serviceLinks.preprintsSupport;
+    @tracked selectedProvider: PreprintProviderModel | undefined;
+    @tracked selectedProviderId: string | undefined;
```

Comment:
Currently, WE update both `selectedProvider` and `selectedProviderId` upon provider selection. But this is not necessary as you can easily infer one from another. If we really need to know the `id` of the selected provider, a getter like this will do:
```
get selectedProviderId() {
  return this.selectedProvider.id;
}
```